pytorch_lightning/trainer/trainer.py

```diff
@@ -1550,13 +1550,17 @@ def _log_device_info(self) -> None:
                 " `Trainer(ipus=8)` or script `--ipus=8`."
             )
 
-    def _on_exception(self):
+    def _on_exception(self, exception: Exception) -> None:
         if not _fault_tolerant_training():
             return
         # save a checkpoint for fault tolerant training. we don't use `log_dir` to minimize the chances of failure.
         file_path = os.path.join(self.default_root_dir, ".pl_auto_save.ckpt")
         self.save_checkpoint(file_path)
 
+        # this is required to ensure schedulers can properly restart the run.
+        if isinstance(exception, ExitGracefullyException):
+            os._exit(0)
```

Comment:
why not `raise SystemExit(0)` or `exit(0)`?

Reply:
Or even better, we can have `ExitGracefullyException` subclass `SystemExit`

Reply:
Done !